src/uu/install/src/install.rs

```diff
@@ -626,6 +653,44 @@ fn copy_files_into_dir(files: &[PathBuf], target_dir: &Path, b: &Behavior) -> UR
     Ok(())
 }
 
+/// Handle incomplete user/group parings for chown.
+///
+/// Returns a Result type with the Err variant containing the error message.
+///
+/// # Parameters
+///
+/// _path_ must exist.
+///
+/// # Errors
+///
+/// If the owner or group are invalid or copy system call fails, we print a verbose error and
+/// return an empty error value.
+///
+fn chown_optional_user_group(path: &Path, b: &Behavior) -> UResult<()> {
+    if b.owner_id.is_some() || b.group_id.is_some() {
+        let meta = match fs::metadata(path) {
+            Ok(meta) => meta,
+            Err(e) => return Err(InstallError::MetadataFailed(e).into()),
+        };
+
+        let verbosity = Verbosity {
+            groups_only: b.owner_id.is_none(),
+            level: VerbosityLevel::Normal,
+        };
+
+        match wrap_chown(path, &meta, b.owner_id, b.group_id, false, verbosity) {
+            Ok(n) => {
+                if !n.is_empty() {
+                    return Err(InstallError::ChownFailed(path.to_path_buf(), n).into());
+                }
+            }
+            Err(e) => return Err(InstallError::ChownFailed(path.to_path_buf(), e).into()),
+        }
```

Comment:
Nit: This can be condensed a little:
```suggestion
        match wrap_chown(path, &meta, b.owner_id, b.group_id, false, verbosity) {
            Ok(n) if n.is_empty()  => {},
            Ok(e) | Err(e) => return Err(InstallError::ChownFailed(path.to_path_buf(), e).into()),
        }
```

More importantly, however, why are you treating the `Ok` case as an error? Maybe it should just be printed right here? I think your current implementation will set the exit code to 1 on a successful chown, which doesn't seem correct.

Reply:
Tried to leave this as it was, but it doesn't make sense as an error. The main driver here for `wrap_chown()` printing informative messages is the verbosity passed in which is `VerbosityLevel::Normal`.  However, the current implementation of `wrap_chown()` never returns any strings.

When comparing with GNU `install`, it doesn't print `chown` operations with or with the verbose flag.

I updated it to be more explicit and `println!(msg)` if `b.verbose` and `wrap_chown()` returns a string in the future.

Please take a look now.

I might argue we should disregard all verbose `wrap_chown()` output to better mimick GNU Coreutils, but mostly trying to maintain previous behavior so I don't break something else that's not tested.

Reply:
> When comparing with GNU install, it doesn't print chown operations with or with the verbose flag.

Oh my bad then. You're absolutely right: in that case we should be compatible and not print anything.

Reply:
Currently it doesn't print anything unless `wrap_chown` is changed and that'd be a bug there with mishandling the verbosity.

This is ready for merge if you're happy.